File: unicode_encoder_cli.rb
# Clear the screen.
system('cls') || system('clear')

# Ask the user to input text to encode.
print "Input text to encode: "
text = gets.chomp

# Encode the inputted text using Unicode encoding.
unicode_enc_text = text.each_codepoint.map { |cp| "\\u#{cp.to_s(16).rjust(4, '0')}" }.join(' ')

# Print the Unicode encoded text.
puts "Unicode encoded text: #{unicode_enc_text}"

File: unicode_encoder_cli.py
import os

# Clear the screen.
os.system('cls' if os.name == 'nt' else 'clear')

# Ask the user to input text to encode.
text = input("Input text to encode: ")

# Encode the inputted text using Unicode encoding.
unicode_enc_text = ' '.join(['\\u{:04x}'.format(ord(c)) for c in text])

# Remove the BOM (Byte Order Mark) and null characters at the beginning of the byte sequence from output.
unicode_enc_text_clean = unicode_enc_text.replace('\\ufeff', '').replace('\\u0000', '')

# Print the Unicode encoded text cleaned from BOM and null characters.
print(f"Unicode encoded text: {unicode_enc_text_clean}")